app/charts/chart-loading-wrapper.tsx

```diff
@@ -83,19 +92,20 @@ export const ChartLoadingWrapper = <
   if (metadata && dimensions && measures && observations) {
     const { title } = metadata;
 
-    return observations.length > 0 ? (
+    return (
       <Box
-        data-chart-loaded={
-          !(fetchingMetadata && fetchingComponents && fetchingObservations)
-        }
+        data-chart-loaded={!chartLoadingState.loading}
```

Comment:
I wonder if we could not simplify the chartLoadingState logic by just having a context provider at the ChartLoadingWrapper level. This way we could remove the chart state observable ?

```jsx
<ChartLoadingProvider value={{ fetching }}>
      <Box
        data-chart-loaded={!chartLoadingState.loading}
		...
	...
</ChartLoadingProvider>
```

Reply:
I also thought about this, but unfortunately we need to access the loading state "above" the chart, in interactive filters :( That's why I've moved it to the nearest closest parent of both. Tell me if I miss something 😅